pkg/scalers/azure_pipelines_scaler.go

```diff
@@ -0,0 +1,185 @@
+		meta.organizationURL = config.ResolvedEnv[val]
+	} else {
+		return nil, fmt.Errorf("no organizationURL given")
+	}
+
+	if val, ok := config.AuthParams["personalAccessToken"]; ok && val != "" {
+		// Found the personalAccessToken in a parameter from TriggerAuthentication
+		meta.personalAccessToken = config.AuthParams["personalAccessToken"]
+	} else if val, ok := config.TriggerMetadata["personalAccessTokenFromEnv"]; ok && val != "" {
+		meta.personalAccessToken = config.ResolvedEnv[config.TriggerMetadata["personalAccessTokenFromEnv"]]
+	} else {
+		return nil, fmt.Errorf("no personalAccessToken given")
+	}
+
+	if val, ok := config.TriggerMetadata["poolID"]; ok && val != "" {
+		meta.poolID = val
+	} else {
+		return nil, fmt.Errorf("no poolID given")
+	}
+
+	return &meta, nil
+}
+
+func (s *azurePipelinesScaler) GetMetrics(ctx context.Context, metricName string, metricSelector labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
+	queuelen, err := s.GetAzurePipelinesQueueLength(ctx)
+
+	if err != nil {
+		azurePipelinesLog.Error(err, "error getting pipelines queue length")
+		return []external_metrics.ExternalMetricValue{}, err
+	}
+
+	metric := external_metrics.ExternalMetricValue{
+		MetricName: metricName,
+		Value:      *resource.NewQuantity(int64(queuelen), resource.DecimalSI),
+		Timestamp:  metav1.Now(),
+	}
+
+	return append([]external_metrics.ExternalMetricValue{}, metric), nil
+}
+
+func (s *azurePipelinesScaler) GetAzurePipelinesQueueLength(ctx context.Context) (int, error) {
+	url := fmt.Sprintf("%s/_apis/distributedtask/pools/%s/jobrequests", s.metadata.organizationURL, s.metadata.poolID)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return -1, err
+	}
+
+	req.SetBasicAuth("PAT", s.metadata.personalAccessToken)
+
+	r, err := s.httpClient.Do(req)
+	if err != nil {
+		return -1, err
+	}
+
+	b, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return -1, err
+	}
+	r.Body.Close()
+
+	if !(r.StatusCode >= 200 && r.StatusCode <= 299) {
+		return -1, fmt.Errorf("azure Devops REST api returned error. status: %d response: %s", r.StatusCode, string(b))
+	}
+
+	var result map[string]interface{}
+	err = json.Unmarshal(b, &result)
+	if err != nil {
+		return -1, err
+	}
+
+	var count int = 0
+	jobs, ok := result["value"].([]interface{})
+
+	if !ok {
+		return -1, fmt.Errorf("api result returned no value data")
+	}
+
+	for _, value := range jobs {
+		v := value.(map[string]interface{})
+		if v["result"] == nil {
+			count++
+		}
+	}
+
+	return count, err
+}
+
+func (s *azurePipelinesScaler) GetMetricSpecForScaling() []v2beta2.MetricSpec {
+	targetPipelinesQueueLengthQty := resource.NewQuantity(int64(s.metadata.targetPipelinesQueueLength), resource.DecimalSI)
+	externalMetric := &v2beta2.ExternalMetricSource{
+		Metric: v2beta2.MetricIdentifier{
+			Name: kedautil.NormalizeString(fmt.Sprintf("%s-%s", "azure-pipelines-queue", s.metadata.poolID)),
```

Comment:
`s.metadata.poolID` - is the poolID unique on Azure org level or is it unique across the whole Azure platform? We need to have the generated metricName unique enough, so if there are two triggers in the same ScaledObject, we can distinguish them by the metricName. 

So here comes the hypotetical example, can we have two Azure Pipelines triggers in the same ScaledObject that both have the same `poolID` but are in different Azure org?  -> Which will result in the same `metricName` for both triggers, but practically they are pointing to different Azure Pipelines.

Reply:
You are correct, this should be more unique. I will change this so the organization name is included as well.